aips/aip-103.md

```diff
@@ -0,0 +1,206 @@
+
+   Since the `aud` value is committed to in the `IdCommitment`, a proof generated with `skip_aud_check` will be rejected by the validator if the account is not `aud`-less, as the `IdCommitment` will be different, resulting in proof verification failing due to the public inputs hash being incorrect.
+
+   Similarly, a proof generated without `skip_aud_check` will be rejected by the validator if the account *is* `aud`-less, as in this case as well the `IdCommitment` will be different, resulting in proof verification failing due to the public inputs hash being incorrect.
+
+2. The prover API will also require an update to allow for indiciating whether the `aud` check is enabled.  This will be done by adding a new boolean argument `skip_aud_check` to the `prove` API.
+   
+   ```rust
+   #[derive(Debug, Serialize, Deserialize)]
+   pub struct RequestInput {
+      pub jwt_b64: String,
+      pub epk: EphemeralPublicKey,
+      #[serde(with = "hex")]
+      pub epk_blinder: EphemeralPublicKeyBlinder,
+      pub exp_date_secs: u64,
+      pub exp_horizon_secs: u64,
+      pub pepper: Pepper,
+      pub uid_key: String,
+      pub extra_field: Option<String>,
+      pub aud_override: Option<String>,
+      pub skip_aud_check: bool, // New argument
+   }
+   ```
+
+3. The pepper API will also require an update to allow for indiciating whether the `aud` check is enabled.  This is because for the pepper is derived using the `aud` value as one of the inputs, and for audless accounts the `aud` value needs to be the empty string in order for the account to be used across applications (which will differ by the value of the `aud` claim in the JWT).  Thus the pepper API needs to know whether the account is `aud`-less to construct the pepper appropriately.  This will be done by adding a new boolean argument `skip_aud_check` to the `fetch_pepper` API.
+
+   ```rust
+   #[derive(Debug, Deserialize, Serialize)]
+   pub struct PepperRequest {
+      pub jwt: String,
+      pub epk: EphemeralPublicKey,
+      pub exp_date_secs: u64,
+      pub epk_blinder: Vec<u8>,
+      pub uid_key: Option<String>,
+      pub derivation_path: Option<String>,
+      pub skip_aud_check: bool, // New argument
+   }
+   ```
+
+4. The SDK will also need to be updated to support instantiating of such accounts.  This will require adding a new boolean argument to the `KeylessAccount` constructor and constructing the `KeylessPublicKey` and `AccountAddress` appropriately.
+
+## Testing (Optional)
+
+1. Write unit tests for the circuit to verify that it correctly handles the `aud` check.
+2. Write unit tests for the SDK to verify that it correctly instantiates accounts with and without `aud` checks.
+3. Do a manual end-to-end test in devnet/testnet via the SDK once the verification key is updated.
+4. Write smoke tests ensuring that `aud`-less accounts are rejected if the account requires `aud` to be present.
```

Comment:
5. Make sure pepper service and prover service rejects `skip_aud_check = true` for non-audless OIDC providers?